apps/extension/background-script.ts

```diff
@@ -251,14 +246,34 @@ runtime.onConnect.addListener(port => {
 });
 
 // listen messages from UI
-runtime.onMessage.addListener((message: BackgroundMessage) => {
+runtime.onMessage.addListener(async (message: BackgroundMessage) => {
   switch (message.type) {
     case BackgroundMessageType.SetPasswordHash: {
-      passwordHash = message.data.passwordHash;
+      storage.local.set({ [LOGIN_TOKEN]: message.data.passwordHash });
+
+      return Promise.resolve();
+    }
+    case BackgroundMessageType.SetLockTimePeriod: {
+      storage.local.set({ [LOCK_TIME_PERIOD]: message.data.lockTimePeriod });
 
       return Promise.resolve();
     }
     case BackgroundMessageType.GetPasswordHash: {
+      const passwordHash: string | null = await fetchFromStorage(LOGIN_TOKEN);
+      const lastUserActivity = await fetchFromStorage(LAST_USER_ACTIVITY_TIMESTAMP);
+      const lockTimePeriod: number | null = await fetchFromStorage(LOCK_TIME_PERIOD);
+
+      if (
+        Date.now() > lastUserActivity + convertLockTime(lockTimePeriod ?? 1) &&
+        !isFullpageOpen &&
+        passwordHash !== null &&
+        passwordHash.length
+      ) {
+        storage.local.set({ [LOGIN_TOKEN]: INITIAL_PASSWORD_HASH, [LAST_USER_ACTIVITY_TIMESTAMP]: 0 });
+
+        return Promise.resolve('');
+      }
+
```

Comment:
`background-script` should be as clean as possible.  

create `apps/extension/src/utils/session.utils.ts` and move all logic there.
Then `background-script` will look something like this 
```
runtime.onMessage.addListener(message => {
    switch (message.type) {
        case BackgroundMessageType.SetPasswordHash:
            return setSessionPaswordHash();
        case BackgroundMessageType.GetPasswordHash:
            return getSessionPasswordHash();
......
    }
})
```